components/leo/BUILD.gn

```diff
@@ -6,6 +6,10 @@
 import("//brave/components/common/typescript.gni")
 
 transpile_web_ui("leo") {
+  output_dir = "$root_gen_dir/brave/ui/webui/resources/leo/js"
```

Comment:
This stands out as something a bit off. If it's only meant to be generated directly at that path, perhaps this shouldn't be a separate components, and should live in `brave/ui/webui/resources/BUILD.gn`? It seems like the way it only outputs to that dependency target's gen dir means that it's not intended to be a dependency for any other target in the project, is that right?
Or will we want multiple targets to consume and serve the GRD(P) themselves? If so, maybe we need to come up with an alternative...

Reply:
Okay, so I didn't actually need this :laughing: I've removed it and it still works fine. I have one final problem, with the typechecking stuff:

Basically, in the generated code it tries to load `gen/brave/ui/webui/resources/tsc/leo.bundle.js` because that's where it looks for all generated files, and that doesn't exist. It's only a Typescript error, so if I touch that file everything works fine.

I don't really want to copy files to that folder but it also seems a bit painful updating the `tsconfig` to look for the leo bundle in the right place. Do you have any ideas?

Reply:
Actually, this isn't a problem! I hadn't applied patches - the build works fine!